### PR TITLE
chore: simplify contributor guide

### PR DESCRIPTION
**Pull request by dimaqq**
https://warthogs.atlassian.net/browse/CHARMTECH-430

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -5,80 +5,28 @@ Thank you for investing your time in contributing to our project! :confetti_ball
...
-- Fork the project into your account.
-- Clone your fork.
-- Add an upstream remote: `git remote add upstream git@github.com:juju/python-libjuju.git`
-- Check with `git remote -v`, origin should point to your fork, upstream should point to ours.
-- If your changes are going to be on top of a specific branch (e.g., `2.9`), then fetch and switch to that.
-- `git switch -c your-branch-name`
-- Make sure that you have run `pre-commit install`
-- Make your changes, create your commits.
-- `git push -u origin your-branch-name`
+- Commit titles have to follow https://www.conventionalcommits.org/.
```
Comment (james-garner-canonical) [inline docs/CONTRIBUTING.md]:
```suggestion
Commit titles must follow [conventional commits](https://www.conventionalcommits.org), with the Juju project's [conventional commit types](https://github.com/juju/juju/blob/main/doc/conventional-commits.md)
```

**Review by james-garner-canonical — APPROVED** [review]
Looks great, just one small suggestion to link to the commit types we use, since conventionalcommits.org itself doesn't care about the exact types, but python-libjuju's CI certainly does

File: docs/CONTRIBUTING.md (hunk 1 above)
Comment (dimaqq) [reply docs/CONTRIBUTING.md]: done

**Pull request merged**